libafl/src/feedbacks/map.rs

```diff
@@ -590,79 +593,83 @@ where
+        debug_assert!(len >= size);
 
-        let history_map = map_state.history_map.as_slice();
+        let mut offset = 0;
+        for map in observer.as_slice_iter() {
+            let map = map.as_slice();
+            let history_map = &map_state.history_map.as_slice()[offset..];
 
-        // Non vector implementation for reference
-        /*for (i, history) in history_map.iter_mut().enumerate() {
-            let item = map[i];
-            let reduced = MaxReducer::reduce(*history, item);
-            if DifferentIsNovel::is_novel(*history, reduced) {
-                *history = reduced;
-                interesting = true;
-                if self.novelties.is_some() {
-                    self.novelties.as_mut().unwrap().push(i);
+            // Non vector implementation for reference
+            /*for (i, history) in history_map.iter_mut().enumerate() {
+                let item = map[i];
+                let reduced = MaxReducer::reduce(*history, item);
+                if DifferentIsNovel::is_novel(*history, reduced) {
+                    *history = reduced;
+                    interesting = true;
+                    if self.novelties.is_some() {
+                        self.novelties.as_mut().unwrap().push(i);
+                    }
                 }
-            }
-        }*/
+            }*/
 
-        let steps = size / VectorType::LEN;
-        let left = size % VectorType::LEN;
+            let steps = size / VectorType::LEN;
+            let left = size % VectorType::LEN;
 
-        if let Some(novelties) = self.novelties.as_mut() {
-            novelties.clear();
-            for step in 0..steps {
-                let i = step * VectorType::LEN;
-                let history = VectorType::from_slice(&history_map[i..]);
-                let items = VectorType::from_slice(&map[i..]);
+            if let Some(novelties) = self.novelties.as_mut() {
+                novelties.clear();
+                for step in 0..steps {
+                    let i = step * VectorType::LEN;
+                    let history = VectorType::from_slice(&history_map[i..]);
+                    let items = VectorType::from_slice(&map[i..]);
 
-                if items.simd_max(history) != history {
-                    interesting = true;
-                    unsafe {
-                        for j in i..(i + VectorType::LEN) {
-                            let item = *map.get_unchecked(j);
-                            if item > *history_map.get_unchecked(j) {
-                                novelties.push(j);
+                    if items.simd_max(history) != history {
+                        interesting = true;
+                        unsafe {
+                            for j in i..(i + VectorType::LEN) {
+                                let item = *map.get_unchecked(j);
+                                if item > *history_map.get_unchecked(j) {
+                                    novelties.push(j + offset);
```

Comment:
This is the major change.

If IntoIter is Once, this should optimise out the `offset` and `for` shenanigans.